pkg/services/internal/whatsonchain/service.go

```diff
@@ -0,0 +1,79 @@
+package whatsonchain
+
+import (
+	"fmt"
+	"log/slog"
+	"net/http"
+	"time"
+
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/defs"
+	"github.com/go-resty/resty/v2"
+	"github.com/go-softwarelab/common/pkg/to"
+)
+
+type WhatsOnChain struct {
+	httpClient *resty.Client
+	url        string
+	apiKey     string
+	logger     *slog.Logger
+}
+
+func New(httpClient *resty.Client, logger *slog.Logger, apiKey string, network defs.BSVNetwork) *WhatsOnChain {
+	return &WhatsOnChain{
+		httpClient: httpClient,
+		apiKey:     apiKey,
+		url:        fmt.Sprintf("https://api.whatsonchain.com/v1/bsv/%s", network),
+		logger:     logger,
+	}
+}
+
+func (woc *WhatsOnChain) UpdateBsvExchangeRate(exchangeRate *BSVExchangeRate, bsvUpdateMsecs *int) (BSVExchangeRate, error) {
+	if exchangeRate != nil {
+		fifteenMinutesMs := 1000 * 60 * 15
```

Comment:
it's rather a const.

BTW. you could make it clearer by calculating it like that:
```suggestion
		fifteenMinutesMs := 15 * time.Minute.Milliseconds()
```

Reply:
But then it's not a duration, we can have `15 * time.Minute` but then it's not a miliseconds, but I'm not sure if it even needs to be ms. Maybe it doesn't